dimagi/utils/couch/migration.py

```diff
@@ -135,10 +136,10 @@ def _migration_get_couch_model_class(cls):
         raise NotImplementedError()
 
     def _migration_get_couch_object(self):
-        if not self.couch_id:
+        if not getattr(self, self.couch_id_name, None):
```

Comment:
Good point about the hard failure.  I also agree that a name beginning with `_migration_` would be better, will update.  Why do you guys advocate for a method over a class attr?  I can't imagine we'd want something dynamic.